event/event.go

```diff
@@ -42,9 +43,26 @@ type (
 	// by using [slog.FromCtx].
 	Handler[T any] func(context.Context, T) error
 
+	// HandlerWithMetadata is responsible for handling events from a [Subscription] with the associated event metadata.
+	// The context passed to the handler will have all common metadata also passed to calls to [Handler] and extra metadata
+	// that is more specific as defined on [Metadata].
```

Comment:
Is that correct? I don't see where the extra metadata is added to the context.

Reply:
It is..but it is confusing. I'm referencing the trace ID and the org ID as metadata...because they are :laughing: ...but yeah we have two sets of metadata...the general ones and the too specific ones (that you pointed out before). Hmm need to explain this better here.

Reply:
Tried to improve the docs =P